llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp

```diff
@@ -3741,9 +3741,11 @@ LegalizerHelper::lower(MachineInstr &MI, unsigned TypeIdx, LLT LowerHintTy) {
   }
   case TargetOpcode::G_ATOMIC_CMPXCHG_WITH_SUCCESS: {
     auto [OldValRes, SuccessRes, Addr, CmpVal, NewVal] = MI.getFirst5Regs();
-    MIRBuilder.buildAtomicCmpXchg(OldValRes, Addr, CmpVal, NewVal,
+    auto Tmp = MRI.createGenericVirtualRegister(MRI.getType(OldValRes));
+    MIRBuilder.buildAtomicCmpXchg(Tmp, Addr, CmpVal, NewVal,
                                   **MI.memoperands_begin());
-    MIRBuilder.buildICmp(CmpInst::ICMP_EQ, SuccessRes, OldValRes, CmpVal);
+    MIRBuilder.buildCopy(OldValRes, Tmp);
```

Comment:
I'd expect to be able to pass both dest registers to MIRBuilder.buildAtomicCmpXchg

Reply:
Both? What is the other one for `MIRBuilder.buildAtomicCmpXchg`?

Reply:
for G_ATOMIC_CMPXCHG_WITH_SUCCESS there are 2 outputs, which I thought was in use for some reason

Reply:
Yeah but I think yeah it is lowering `G_ATOMIC_CMPXCHG_WITH_SUCCESS` to `G_ATOMIC_CMPXCHG` followed by a `ICmp`.